### PR TITLE
Add search by tags, random pick popular recipes

### PR DESCRIPTION
**Pull request by shadow-blade-X**
Now the function allows search by tags and random pick popular recipes if nothing shows up

### CODE REVIEW

File: PlateUp-flask/run.py [hunk 1]
```diff
@@ -336,10 +342,10 @@ def get(self):
         limit=int(request.args.get('Limit')) if request.args.get('Limit') else 20
         page=int(request.args.get('Page')) if request.args.get('Page') else 0
 
-        #if self.__debug==True:
```
Comment (UTkzhang) [inline PlateUp-flask/run.py:339]: comment out the debug stuff for commits

File: PlateUp-flask/run.py [hunk 2]
```diff
@@ -350,16 +356,16 @@ def get(self):
         if search_query!=None:
             recipe_list_name=self.__search_for_recipes_by_name(search_query)
             recipe_list_ingredient=self.__search_for_recipes_by_ingredient(search_query)
+            recipe_list_tags=self.__search_for_recipes_by_tags(search_query)
 
         recipe_list=self.__merge_list(recipe_list_name, recipe_list_ingredient)
+        recipe_list=self.__merge_list(recipe_list, recipe_list_tags)
```
Comment (UTkzhang) [inline PlateUp-flask/run.py:366]: can we do hashtable merge here? this seems quite inefficient

Comment (UTkzhang) [reply PlateUp-flask/run.py:366]: also remove the comment regarding hash I left

File: PlateUp-flask/run.py [hunk 3]
```diff
@@ -186,6 +186,10 @@ def __search_in_database_by_keyword_name(self, keywords):
         recipe_found = db.session.query(Recipe).filter(Recipe.name.like(keywords)).all()
         return recipe_found
 
+    def __search_in_database_by_keyword_tag(self, keywords):
+        recipe_found = db.session.query(Recipe).filter(Recipe.tags.like(keywords)).all()
```
Comment (UTkzhang) [inline PlateUp-flask/run.py]: keyword not keywords right (only support 1 term?)

how do you deal with capitalization? or when user inputs gluten free and you want to return glutenFree tag?

**Review by UTkzhang — CHANGES_REQUESTED** [review]

File: PlateUp-flask/run.py (hunk 2 above)
Comment (shadow-blade-X) [reply PlateUp-flask/run.py:366]: Change the list to dictionaries will need to change too much. Let's add the hash table merge to a optimization problem in the future.

Comment (UTkzhang) [reply PlateUp-flask/run.py:366]: kk

**Review by UTkzhang — APPROVED** [review]

**Pull request merged**